package/Runtime/Scripts/Internal/IOSAdManager.cs

```diff
@@ -0,0 +1,63 @@
+﻿using UnityEngine;
+
+namespace Nimbus.Runtime.Scripts.Internal
+{
+    public class IOSAdManager : MonoBehaviour
```

Comment:
You can't do this, you are creating a new game object in the hierarchy. There exist 1 prefab which is mandatory and explicitly dropped by the N user into the game hierarchy . That is the Nimbus manager. It is also the only public facing API. Voodoo has made it clear they do not wanted added Game objects because when they hand off their SDK for the multitude publishers, they cannot actively change the hierarchy. In the case with the Nimbus manager they ads are manually attaching it as a child of their object. Here you are creating an object without setting the parent and you can't know the parent object.  I do not do this for the Android implementation.